userspace/libsinsp/filter/ast.h

```diff
@@ -181,24 +187,40 @@ struct SINSP_PUBLIC or_expr: expr
     bool is_equal(const expr* other) const override
     {
         auto o = dynamic_cast<const or_expr*>(other);
-        return o != nullptr && std::equal(
-            children.begin(), children.end(), 
-            o->children.begin(), compare);
+        if (o == nullptr)
+        {
+            return false;
+        }
+
+        for (size_t i = 0; i < children.size(); i++)
+        {
+            if (o->children.size() <= i)
+            {
+                return false;
+            }
+
+            if (!compare(children[i].get(), o->children[i].get()))
+            {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
This is not as clever and elegant as before, but I think it makes sense in this case to be completely explicit about the pointer usage.